flag_test.go

```diff
@@ -859,16 +847,21 @@ func TestInt64FlagWithEnvVarHelpOutput(t *testing.T) {
 		fl := IntFlag{Name: test.name, EnvVars: []string{"APP_BAR"}}
 		output := fl.String()
 
-		expectedSuffix := " [$APP_BAR]"
-		if runtime.GOOS == "windows" {
-			expectedSuffix = " [%APP_BAR%]"
-		}
+		expectedSuffix := suffixForEnv("APP_BAR")
 		if !strings.HasSuffix(output, expectedSuffix) {
 			t.Errorf("%s does not end with"+expectedSuffix, output)
 		}
 	}
 }
 
+func suffixForEnv(s string) string {
+	expectedSuffix := fmt.Sprintf(" [$%s]", s)
+	if runtime.GOOS == "windows" {
```

Comment:
can we do this with [constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints)? Like, having a file like `suffix_windows.go`:

```go
//go:build windows
package cli

import "fmt"

func suffixForEnv(s string) string {
        return fmt.Sprintf(" [%%%s%%]", s)
}
```

and another one called like `suffix_nonwindows.go`:

```go
//go:build !windows
package cli

import "fmt"

func suffixForEnv(s string) string {
        return fmt.Sprintf(" [$%s]", s)
}
```

Reply:
Interesting. Yes we could do that. However I found a function in the code which does the suffix stuff already

Reply:
Worth considering how this relates to https://github.com/urfave/cli/issues/1288 ?